content/articles/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/index.md

```diff
@@ -0,0 +1,248 @@
+
+```bash
+npm install -D @types/node
+```
+
+Node.js packages are written in JavaScript and not Typescript. To get the type definitions for its packages, you need to install third-party packages called `@types`. For example, to use [Express type definations](https://www.npmjs.com/package/@types/express), install `@types/express` by running:
+
+```bash
+npm install -D @types/express
+```
+
+This will give you access to type definitions for Express.
+
+Not all packages have `@types`. Package with types has the following NPM `@types` tag.
+
+![Npm types declarations](/engineering-education/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/npm-types-declarations.png)
+
+Let's set a simple Express server using Typescript.
+
+- Import the package. When using Node.js, you would typically import a package using the `require()` function. With Typescript, this is a little different. Use the keyword `import` to access a package module, as shown below.
+
+```ts
+import express, {Request,Response,Application} from 'express';
+```
+
+- Initialize Express inside Typescript.
+
+```ts
+const app:Application = express();
+```
+
+- Set the server port.
+
+```ts
+const PORT = process.env.PORT || 8000;
+```
+
+- Set an end point/route.
+
+```ts
+app.get("/", (req:Request, res:Response):void => {
+  res.send("Hello Typescript with Node.js!")
+});
+```
+
+- Listen to the server port.
+
+```ts
+app.listen(PORT, ():void => {
+  console.log(`Server Running here 👉 https://localhost:${PORT}`);
+});
+```
+
+- Run `npm start` to set the server up and running. And when you open `https://localhost:8000/` in your browser, a response will be sent back to you as defined in the `res.send()`.
+
+When developing an extensive application, it's advisable to hook your project with watch parameters that will help you restart your server whenever you make and saves changes to your code structure. There are two main options:
+
+1. Ts-node-dev - [Ts-node-dev](https://www.npmjs.com/package/ts-node-dev) basically watches `.ts` files, and whenever you make a change, it will restart the server for you. 
+
+  Run the below command to install it. 
+
+  ```bash
+  npm install -D ts-node-dev
+  ```
+
+  To use it, modify the `package.json` script tag as shown below.
+
+  ```json
+  "scripts": {
+    "dev": "ts-node-dev --respawn ./src/index.ts"
+  }
+  ```
+
+  Then run `npm run dev`, and your server will be watched every time you make changes in your code.
+
+![Ts-node-dev](/engineering-education/a-step-by-step-guide-on-how-to-use-typescript-with-nodejs/ts-node-dev.png)
+
+2. Nodemon - [Nodemon](https://www.npmjs.com/package/nodemon) works the same as Ts-node-dev.
+
+  Run the below command to install it. 
+
+  ```bash
+  npm install -D nodemon
+  ```
+  
+  To use it in your project, edit the `package.json` script tag as follows:
+  
+  ```json
+  "scripts": {
+    "dev": "nodemon ./src/index.ts"
+  }
+  ```
+
+  Then run `npm run dev`.
```

Comment:
Could you briefly explain what's the difference between the two packages?

Reply:
> Could you briefly explain what's the difference between the two packages?

I have added a line explaining Nodemon.